protocol/communities/manager.go

```diff
@@ -1894,58 +1896,81 @@ func (m *Manager) CreateHistoryArchiveTorrent(communityID types.HexBytes, msgs [
 			continue
 		}
 
-		wakuMessageArchive := m.createWakuMessageArchive(from, to, messages, topicsAsByteArrays)
-		encodedArchive, err := proto.Marshal(wakuMessageArchive)
-		if err != nil {
-			return archiveIDs, err
+		// Not only do we partition messages, we also chunk them
+		// roughly by size, such that each chunk will not exceed a given
+		// size and archive data doesn't get too big
+		messageChunks := make([][]types.Message, 0)
+		currentChunkSize := 0
+		currentChunk := make([]types.Message, 0)
+		for i, msg := range messages {
+			msgSize := len(msg.Payload) + len(msg.Sig)
+			if currentChunkSize+msgSize > maxArchiveSizeInBytes && len(currentChunk) > 0 {
```

Comment:
I see that this `len(currentChunk) > 0` check makes sense only in the first iteration, when the first message is already over `maxArchiveSizeInBytes`, in which case we violate the constraint and add it anyway.

Isn't going over `maxArchiveSizeInBytes` if we include only one message an issue?

In the second iteration onwards, there will always be something in the `currentChunk`.

Reply:
After you've pointed that out, I decided to drop this altogether and simply drop messages that are already bigger than our max cap.
Please have another look!